custom-payment-flow/client/ios-swiftui/AcceptAPayment/Views/ApplePay.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ApplePay.swift
+//  AcceptAPayment
+//
+//  Created by Thorsten Schaeff on 6/25/21.
+//
+
+import SwiftUI
+import Stripe
+
+struct ApplePay: View {
+    @ObservedObject var backendModel = BackendModel()
+    @StateObject var applePayModel = ApplePayModel()
+    
+    var body: some View {
+        VStack {
+            if backendModel.paymentIntentParams != nil {
+                PaymentButton() {
+                    applePayModel.pay(clientSecret: backendModel.paymentIntentParams?.clientSecret)
+                }
+                .padding()
+            } else {
+                Text("Loading...")
+            }
+            if let paymentStatus = applePayModel.paymentStatus {
+                HStack {
+                    switch paymentStatus {
+                    case .success:
+                        Image(systemName: "checkmark.circle.fill").foregroundColor(.green)
+                        Text("Payment complete!")
+                    case .error:
+                        Image(systemName: "xmark.octagon.fill").foregroundColor(.red)
+                        Text("Payment failed!")
+                    case .userCancellation:
+                        Image(systemName: "xmark.octagon.fill").foregroundColor(.orange)
+                        Text("Payment canceled.")
+                    @unknown default:
+                        Text("Unknown status")
+                    }
+                }
+            }
+        }.onAppear {
+            if (!StripeAPI.deviceSupportsApplePay()) {
+                print("Either Apple Pay is not set up on this device or your configuration is incomplete.")
```

Comment:
nit: This should be "Apple Pay is not supported on this device" — this will return `true` if Apple Pay works but a card is not provisioned, and (I think) tapping the Apple Pay button will guide the user through setting up Apple Pay.

Reply:
(Though our own docs are wrong and Apple's are ambiguous, so I'll go file a task to test this and fix the docs.)